### PR TITLE
Criterion delete issue

### PR DESCRIPTION
**Pull request by JoelMcKinnon**
Changed inputCount to a computed property. This way the render always updates based on the selected operation. This resolves the issue of an extra input appearing without an operation being selected in the criterion remaining after deletion. It was also necessary to add this.checkTelemetry() to the computed prop in order to set all select fields properly.

Resolves issue #2828

### CODE REVIEW

File: src/plugins/condition/components/Criterion.vue [hunk 1]
```diff
@@ -162,6 +161,19 @@ export default {
...
+            let inputCount = 0;
+            if (this.criterion.operation) {
+                for (let i = 0; i < this.filteredOps.length; i++) {
+                    if (this.criterion.operation === this.filteredOps[i].name) {
+                        inputCount = this.filteredOps[i].inputCount;
+                        break;
+                    }
+                }
+            }
+            this.checkTelemetry();
```
Comment (davetsay) [inline src/plugins/condition/components/Criterion.vue:175]: A `computed` should never modify any reactive data. `this.checkTelemetry()` will do just that. Why would we need to checkTelemetry when getting inputCount anyways? it should just be providing you with the current state, not modifying it.

File: src/plugins/condition/components/Criterion.vue [hunk 2]
```diff
@@ -276,17 +279,17 @@ export default {
...
                     this.criterion.input = [this.enumerations[0].value.toString()];
                 }
-                this.inputCount = 0;
             }
         },
         persist() {
-            this.$emit('persist', this.criterion);
+            if (this.criterion.telemetry && this.criterion.metadata !== '' && this.criterion.operation !== '') {
+                this.$emit('persist', this.criterion);
+            }
```
Comment (davetsay) [inline src/plugins/condition/components/Criterion.vue:292]: Question. Is this related or just an optimization you spotted while working on this? Also, I don't know the answer to this question, but just making sure if this is the correct check before persisting? I would think if you wanted to not persist, the check would be to see if the state changed, and only persist if it has. This seems to be checking if there is a certain state and only persist in that case. Plus, I don't think criterion.metadata and criterion.operation are ever `''` are they? I think they could only be something, or`undefined`. Maybe `null` if we set it up that way.

**Comment by shefalijoshi** [comment]
@JoelMcKinnon This is not the right solution. The problem here is that they Criterion elements are keyed by the index in Condition.vue.
The solution is to key using a criterion id instead.

File: src/plugins/condition/components/Criterion.vue (hunk 1 above)
Comment (JoelMcKinnon) [reply src/plugins/condition/components/Criterion.vue]: Took this out. With the switch of the index in v-for in Condition to render Critieria it's not needed.

File: src/plugins/condition/components/Criterion.vue (hunk 2 above)
Comment (JoelMcKinnon) [reply src/plugins/condition/components/Criterion.vue]: It was an attempt to solve the problem that was not successful, so removed it.

File: src/plugins/condition/components/Condition.vue [hunk 3]
```diff
@@ -122,7 +122,7 @@
 
             <template v-if="telemetry.length || condition.configuration.criteria.length">
                 <div v-for="(criterion, index) in condition.configuration.criteria"
-                     :key="index"
+                     :key="criterion.telemetry.key"
```
[the remark this answers is not in the document]
Comment (JoelMcKinnon) [reply src/plugins/condition/components/Condition.vue:124]: I did all the changes you suggested, but only criteria.telemetry.key worked for the index. Not sure how I would key off uuid. Tried it in a couple ways but it didn't work.

[the remark this answers is not in the document]
Comment (shefalijoshi) [reply src/plugins/condition/components/Condition.vue:124]: Just use criterion.id here.

File: src/plugins/condition/components/Criterion.vue [hunk 4]
```diff
@@ -162,6 +161,18 @@ export default {
                 }
             }
             return type;
+        },
```
Comment (shefalijoshi) [inline src/plugins/condition/components/Criterion.vue:164]: You shouldn't need this anymore. Let's revert this change and put the other changes back that aren't related to the id

File: src/plugins/condition/ConditionManager.js [hunk 5]
```diff
@@ -108,7 +108,7 @@ export default class ConditionManager extends EventEmitter {
         if (conditionConfiguration) {
             conditionObj = {
                 ...conditionConfiguration,
-                id: uuid(),
+                id: conditionConfiguration.id || uuid(),
```
Comment (shefalijoshi) [inline src/plugins/condition/ConditionManager.js:111]: This change is not needed as this is for the condition, not the criterion.

File: src/plugins/condition/components/Condition.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (JoelMcKinnon) [reply src/plugins/condition/components/Condition.vue]: Done and working!

File: src/plugins/condition/components/Criterion.vue (hunk 4 above)
Comment (JoelMcKinnon) [reply src/plugins/condition/components/Criterion.vue]: done

File: src/plugins/condition/ConditionManager.js (hunk 5 above)
Comment (JoelMcKinnon) [reply src/plugins/condition/ConditionManager.js]: done

**Comment by shefalijoshi** [comment]
Closing and reopening to sync with master.
